micropip/_commands/freeze.py

```diff
@@ -40,12 +40,17 @@ def freeze_data() -> dict[str, Any]:
 
 
 def load_pip_packages() -> Iterator[tuple[str, dict[str, Any]]]:
-    return filter(None, map(load_pip_package, importlib.metadata.distributions()))
+    return map(
+        package_item,
+        filter(None, map(load_pip_package, importlib.metadata.distributions())),
+    )
+
+
+def package_item(entry: dict[str, Any]) -> tuple[str, dict[str, Any]]:
+    return canonicalize_name(entry["name"]), entry
```

Comment:
This is interesting, often the need to write a function like this instead of a lambda dissuades me from using map. The combination of spiral control flow with in line lambdas tends to be a bit unmanageable.

Reply:
Yes. I wouldn't object to inlining this behavior in a generator expression in `load_pip_packages`.